### PR TITLE
[GNA] Fixed scale factors propagation for Eltwise with very different inputs ranges

### PR DESCRIPTION
**Pull request by elilobanova**
### Details:
 - Fixed scale factors propagation for Eltwise with very different inputs ranges
 - Exception is thrown in the case of not positive scale factor
 - Tests are added

### Tickets:
63395

### CODE REVIEW

File: inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp [hunk 1]
```diff
@@ -699,5 +699,52 @@ using QuantI8_I8 = frontend::QuantPair<frontend::QuantI8_I8, frontend::QuantI8_I
...
+    bias
+};
+
+/**
+ * @brief Returns a scale factor for specific layer data
+ * @param layer Layer to be quantized
+ * @param data_type Type of data to be quantized
+ * @return scale factor
+ */
+inline float getScaleFactor(InferenceEngine::CNNLayerPtr layer, QuantizedDataType data_type) {
```
Comment (dmitriikhurtin) [inline inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp:715]: Please add a check for zero for layer.

Comment (elilobanova) [reply inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp:715]: added assert

**Review by dmitriikhurtin — APPROVED** [review]

File: inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp [hunk 2]
```diff
@@ -699,5 +699,53 @@ using QuantI8_I8 = frontend::QuantPair<frontend::QuantI8_I8, frontend::QuantI8_I
...
+            default:
+                THROW_GNA_LAYER_EXCEPTION(layer) << "Unsupported data type for quantization: " << static_cast<int>(data_type);
+        }
+    }
+
+    auto isZero = [](float p1) {
+        return std::abs(p1) <= 0.00001f;
+    };
+
+    if (scale_factor < 0.0 || isZero(scale_factor)) {
```
Comment (sirzabek) [inline inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp:744]: What about infinity? I remember fixing one failing test with adding a check for scale factor reaching infinity.

**Review by sirzabek — APPROVED** [review]

File: inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp (hunk 2 above)
Comment (elilobanova) [reply inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp]: added check for infinity

**Review by nkogteva — APPROVED** [review]

**Pull request merged**